### PR TITLE
Adds new parameter to Clear Index Cache API

### PR DESCRIPTION
**Pull request by hdhalter**
### Description
_Describe what this change achieves._

### Issues Resolved
fixes https://github.com/opensearch-project/documentation-website/issues/4092

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _api-reference/index-apis/clear-index-cache.md [hunk 1]
```diff
@@ -93,6 +93,13 @@ POST /_cache/clear
 ````
 {% include copy-curl.html %}
 
+#### Clear unused entries from the cache on search-capable nodes
+
+````json
```
Comment (Naarcha-AWS) [inline _api-reference/index-apis/clear-index-cache.md:98]:
```suggestion
```json
```

File: _api-reference/index-apis/clear-index-cache.md [hunk 2]
```diff
@@ -93,6 +93,13 @@ POST /_cache/clear
...
+POST http://localhost:9200/*/_cache/clear?file=true 
+````
```
Comment (Naarcha-AWS) [inline _api-reference/index-apis/clear-index-cache.md:100]:
```suggestion
```
```

**Review by Naarcha-AWS — APPROVED** [review]
Removing an extra tick mark from the code example, otherwise, LGTM.

**Comment by hdhalter** [comment]
@kotwanikunal - Hi Kunal, can you please check this for technical accuracy? Thanks.

**Review by kotwanikunal — APPROVED** [review]

**Comment by kotwanikunal** [comment]
> @kotwanikunal - Hi Kunal, can you please check this for technical accuracy? Thanks.

LGTM! Thanks 👍

File: _api-reference/index-apis/clear-index-cache.md [hunk 3]
```diff
@@ -18,7 +18,7 @@ If you use the Security plugin, you must have the `manage index` privileges.
 | Parameter | Data type | Description |
 :--- | :--- | :---
 | target | String | Comma-delimited list of data streams, indexes, and index aliases to which cache clearing will be applied. Wildcard expressions (`*`) are supported. To target all data streams and indexes in a cluster, omit this parameter or use `_all` or `*`. Optional. |
-
+| file | Boolean | If `true`, clears the unused entries from the filecache on 'search' capable nodes. Defaults to `false`. |
```
Comment (natebower) [inline _api-reference/index-apis/clear-index-cache.md:21]: Should it be "file cache" (two words)? As written, it should technically be "`search`-capable nodes" (as in the heading below), but we should avoid that for formatting reasons. Can we rephrase?

**Review by natebower — CHANGES_REQUESTED** [review]
@hdhalter Just a couple of comments to be addressed. Please tag me when complete.

File: _api-reference/index-apis/clear-index-cache.md (hunk 3 above)
Comment (kolchfa-aws) [reply _api-reference/index-apis/clear-index-cache.md:21]: Move line 21 to the Query parameter table. This is not a path parameter.

File: _api-reference/index-apis/clear-index-cache.md [hunk 4]
```diff
@@ -93,6 +93,12 @@ POST /_cache/clear
 ````
 {% include copy-curl.html %}
 
+#### Clear unused entries from the cache on search-capable nodes
+
+```json
+POST http://localhost:9200/*/_cache/clear?file=true
```
Comment (kolchfa-aws) [inline _api-reference/index-apis/clear-index-cache.md:99]: Add three tic marks after the request, otherwise it's not formatted correctly.

Comment (kolchfa-aws) [reply _api-reference/index-apis/clear-index-cache.md:99]: Per API style template, we're not referring to localhost in our requests:
```suggestion
POST /*/_cache/clear?file=true 
```

File: _api-reference/index-apis/clear-index-cache.md [hunk 5]
```diff
@@ -29,8 +29,9 @@ All query parameters are optional.
...
 | fielddata | Boolean | If `true`, clears the fields cache. Use the `fields` parameter to clear specific fields' caches.  Defaults to `true`. |
-| fields | String | Used in conjunction with the `fielddata` parameter. Comma-delimited list of field names that will be cleared out of the cache. Does not support objects or field aliases. Defaults to all fields. |
-| index | String | Comma-delimited list of index names that will be cleared out of the cache. |
+| fields | String | Used in conjunction with the `fielddata` parameter. Comma-delimited list of field names that are cleared out of the cache. Does not support objects or field aliases. Defaults to all fields. |
+| file | Boolean | If `true`, clears the unused entries from the file cache on search-capable nodes. Defaults to `false`. |
```
Comment (kotwanikunal) [inline _api-reference/index-apis/clear-index-cache.md:33]: `search` is a role which is why it was highlighted. We can rephrase it to 
> .. unused entries from the file cache on nodes with `search` role...

**Comment by hdhalter** [comment]
@kotwanikunal - Is there any reason not to call this topic "Clear cache" to keep consistent with the other topics? 

The first sentence says, "The clear cache API operation clears the caches of one or more indexes. For data streams, the API clears the caches of the stream’s backing indexes."

File: _api-reference/index-apis/clear-index-cache.md [hunk 6]
```diff
@@ -29,85 +29,93 @@ All query parameters are optional.
...
 
-````json
+```json
 POST /my-index/_cache/clear?fields=fielda,fieldb
-````
+```
 {% include copy-curl.html %}
 
-##### Clear caches for specific data streams and indexes
+### Clear caches for specific data streams and indexes
```
[the remark this answers is not in the document]
Comment (hdhalter) [reply _api-reference/index-apis/clear-index-cache.md:79]: It clears both.

**Comment by kotwanikunal** [comment]
> @kotwanikunal - Is there any reason not to call this topic "Clear cache" to keep consistent with the other topics?
> 
> The first sentence says, "The clear cache API operation clears the caches of one or more indexes. For data streams, the API clears the caches of the stream’s backing indexes."

File cache is a new and separate entity from the existing cache. We need to use it specifically in this context to make it clear which cache the user intends to clear.

**Comment by hdhalter** [comment]
@kotwanikunal - Would it be correct to say, "The clear cache API operation clears the caches of indexes, data streams, and files. For data streams, the API clears the caches of the stream’s backing indexes."

**Comment by hdhalter** [comment]
Kunal is good with how it is. @Naarcha-AWS - do you want to take another pass? I changed the name of the topic to "Clear cache".

**Review by natebower — APPROVED** [review]
LGTM
